### DISCUSSION TITLE
installing only in home .lv2

### DISCUSSION THREAD

**Discussion by ccaudle** [bruno-unna/euclidean-rhythms#9] [Q&A]
I attempted to build and install to my home .lv2 directory, but meson still prompted for sudo/root password.
I configured with this command:
meson setup --native-file=environments/linux64.ini builddir --libdir=~/.lv2
and the end of the setup step printed this:
  User defined options
    Native files: environments/linux64.ini
    libdir      : ~/.lv2

During the build/install this was still printed, which I was not expecting:
Attempting to use polkit to gain elevated privileges...

I also attempted to configure with this command:
 meson setup --libdir=~/.lv2 builddir
i.e. not using the linux64.ini file, since the only line in that file seemed to be setting libdir to /usr/local/lib which I was overriding on the command line.  Same error about attempting to gain elevated privileges.

Is that an error in the meson  script, or do I need to do something additional to defining the libdir as my home .lv2 directory?

**Comment by ccaudle** [comment]
Found there are two not completely related problems in the meson setup:
[8/8] Linking target test/test_euclidean
Installing src/euclidean.so to /home/chris/.lv2/lv2/euclidean.lv2

I do not believe that extra "lv2" subdirectory should be created, the files should be in ~/.lv2/euclidean.lv2.

And also this:
Installing /home/chris/Src/EuclideanLV2/euclidean-rhythms/include/euclidean.h to **/usr/local/include/**
Installation failed due to insufficient permissions.

Why is meson attempting to copy header files into a system directory?  LV2 plugins should not need to install headers.

Attempting to use polkit to gain elevated privileges...
Error executing command as another user: Not authorized

Running over ssh I did not even get a prompt to enter a password, but I did when compiling in a terminal in a desktop environment.  That may be something in my polkit setup, I am not sure.  Not really related, I just add it here in case someone is familiar with polkit setup, since I would apparently not even be able to build and install into /usr/local when compiling remotely from my machine.

**Reply by bruno-unna** [reply]
You are right. What seems to be the natural value for the flags is not installing the plugin where it needs to be if you want a home-directory installation.

Turns out I'm not an expert on meson (this is the first project for which I use it), so bear with me. I'll do some experimentation, but if you happen to find the solution before I do, please feel free to share.
